extensions-contrib/opentelemetry-emitter/src/test/java/org/apache/druid/emitter/opentelemetry/OpenTelemetryEmitterTest.java

```diff
@@ -0,0 +1,165 @@
+            return null;
+          }
+        };
+
+    OpenTelemetryEmitter emitter = EasyMock.createMock(OpenTelemetryEmitter.class);
+    emitter.emit(notServiceMetricEvent);
+    EasyMock.replay(emitter);
+
+    emitter.emit(notServiceMetricEvent);
+    EasyMock.verifyUnexpectedCalls(emitter);
+  }
+
+  // Check that we don't call "emitQueryTimeEvent" method for ServiceMetricEvent that is not "query/time" type
+  @Test
+  public void testNoEmitNotQueryTimeMetric()
+  {
+    final ServiceMetricEvent notQueryTimeMetric =
+        new ServiceMetricEvent.Builder().build(
+                                            DateTimes.nowUtc(),
+                                            "query/cache/total/hitRate",
+                                            0.54
+                                        )
+                                        .build(
+                                            "broker",
+                                            "brokerHost1"
+                                        );
+
+    OpenTelemetryEmitter emitter = EasyMock.createMock(OpenTelemetryEmitter.class);
+    emitter.emit(notQueryTimeMetric);
+    EasyMock.replay(emitter);
+
+    emitter.emit(notQueryTimeMetric);
+    EasyMock.verifyUnexpectedCalls(emitter);
+  }
+
+  @Test
+  public void testTraceparentId()
+  {
+    final String traceId = "00-54ef39243e3feb12072e0f8a74c1d55a-ad6d5b581d7c29c1-01";
+    final String expectedParentTraceId = "54ef39243e3feb12072e0f8a74c1d55a";
+    final String expectedParentSpanId = "ad6d5b581d7c29c1";
+    final Map<String, String> context = new HashMap<>();
+    context.put("traceparent", traceId);
+
+    final String serviceName = "druid/broker";
+    final DateTime createdTime = DateTimes.nowUtc();
+    final long metricValue = 100;
+
+    NoopExporter noopExporter = new NoopExporter();
+    SdkTracerProvider tracerProvider = SdkTracerProvider.builder()
+                                                        .addSpanProcessor(SimpleSpanProcessor.create(noopExporter))
+                                                        .build();
+
+    final ServiceMetricEvent queryTimeMetric =
+        new ServiceMetricEvent.Builder().setDimension("context", context)
+                                        .build(
+                                            createdTime,
+                                            "query/time",
+                                            metricValue
+                                        )
+                                        .build(
+                                            serviceName,
+                                            "host"
+                                        );
+    OpenTelemetrySdk.builder()
+                    .setTracerProvider(tracerProvider)
+                    .setPropagators(ContextPropagators.create(W3CTraceContextPropagator.getInstance()))
+                    .buildAndRegisterGlobal();
+
+    OpenTelemetryEmitter emitter = new OpenTelemetryEmitter();
```

Comment:
Instead of relying on the global OpenTelemetry in your unit tests, you should change the constructor to take in the `OpenTelemetry` object.  Then call that constructor in your unit test.

```java
  OpenTelemetryEmitter(OpenTelemetry openTelemetry)
  {
    tracer = openTelemetry.getTracer("druid-opentelemetry-extension");
    propagator = openTelemetry.getPropagators().getTextMapPropagator();
  }
```

Then pass in the `OpenTelemetry` object when you crate the `emitter` in the `Module`
```java
public class OpenTelemetryEmitterModule implements DruidModule {
  public Emitter getEmitter(OpenTelemetryEmitterConfig config, ObjectMapper mapper)
  {
    return new OpenTelemetryEmitter(OpenTelemetrySdkAutoConfiguration.initialize(false));
  }
}
```

It's also a good practice to not set the global from this extension using `OpenTelemetrySdkAutoConfiguration.initialize(false)` since there's no need to set the global.